### ISSUE TITLE
[P2] PPO 沒有計入 agari guard 的動作改寫，behaviour denominator 與實際策略不符

### ISSUE THREAD

**Issue by HHim8826** [HHim8826/Mortal#11]
審查分支：`train-parquet`，版本 `08543729000846d56c2602251245596c27f4b182`。原生 agari guard 早已存在；問題是 `ecd2122` 新增 PPO 時把原始 softmax 當成實際出手分布，`75bdb91`／`d7f9dd0` 的 ratio 檢查也無法識別這個差異。

[自我對弈啟用 guard：player.py:322–336](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/mortal/player.py#L322-L336)；[原生動作改寫：libriichi/src/agent/mortal.rs:321–339](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/libriichi/src/agent/mortal.rs#L321-L339)；[PPO 直接計算 softmax likelihood：train_ppo.py:405–413](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/mortal/train_ppo.py#L405-L413)。

在 all-last、最後一名且和牌無法改善名次的狀態，rule_based_agari() 可以否決和牌。若引擎抽到 action 43，原生代理把它改成 logits 最高的非和牌動作。因此替代動作 b 的實際機率應包含 p(b) + p(43)，但 trainer 從 loader 讀到的是最後實際打出的 b，仍使用 log_softmax(...)[b] 計算 mu 和 pi。相同版本的分子／分母一起漏算，所以 ratio=1 的自查照樣通過。

已使用 repo 既有 Rust fixture `rule_based_agari_all_last_minogashi`，在本機真正 libriichi.mjai.Bot 上重播（seat 1，最後摸 8p；可選 8p 棄牌和和牌）：
```text
策略原始機率： P(agari)=0.8, P(discard 8p)=0.2
原始選擇 43 → Bot 最終回傳 dahai 8p
原始選擇 16 → Bot 最終回傳 dahai 8p

實際 P(discard 8p) = 1.0
PPO 使用的 P(discard 8p) = 0.2
```
若新策略的原始機率改成 [agari=.6, discard=.4]，guard 後的棄牌機率仍然是 1，正確 action likelihood ratio 應為 1；目前卻會算成 .4/.2=2，錯誤觸發 clipping 並對不存在的行為差異給梯度。

建議：PPO 的行為策略必須包含這個動作轉換。可記錄實際抽樣的原始 action 並針對該 latent decision 訓練，或在重算 mu／pi 時一致合併 guard 的機率質量；也可明確禁用 PPO rollout 的 guard，再以版本身分區分新舊採樣規則。不能只調整 denominator，分子和 ratio 檢查也須一致。
